app/dashboard/src/layouts/AssetsTable.tsx

```diff
@@ -2256,37 +2254,27 @@ export default function AssetsTable(props: AssetsTableProps) {
             item={item}
             state={state}
             hidden={hidden || visibilities.get(item.key) === Visibility.hidden}
-            selected={isSelected}
-            setSelected={(selected) => {
-              setSelectedKeys(set.withPresence(selectedKeysRef.current, key, selected))
-            }}
-            isSoleSelected={isSoleSelected}
             isKeyboardSelected={
               keyboardSelectedIndex != null && item === visibleItems[keyboardSelectedIndex]
             }
             grabKeyboardFocus={() => {
-              setSelectedKeys(new Set([key]))
+              setSelectedKeys(new Set([item.key]))
               setMostRecentlySelectedIndex(i, true)
             }}
-            allowContextMenu={selectedKeysRef.current.size === 0 || !isSelected || isSoleSelected}
             onClick={onRowClick}
-            onContextMenu={(_innerProps, event) => {
-              if (!isSelected) {
-                event.preventDefault()
-                event.stopPropagation()
-                setMostRecentlySelectedIndex(visibleItems.indexOf(item))
-                selectionStartIndexRef.current = null
-                setSelectedKeys(new Set([key]))
-              }
+            select={() => {
+              setMostRecentlySelectedIndex(visibleItems.indexOf(item))
+              selectionStartIndexRef.current = null
+              setSelectedKeys(new Set([item.key]))
             }}
```

Comment:
Where did this `onContextMenu` behaviour go?

Reply:
inlined into `AssetRow.tsx` because the selected state is now being handled in each row